developers/weaviate/current/retriever-vectorizer-modules/ref2vec-centroid.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: layout-documentation
+solution: weaviate
+sub-menu: Retrievers & Vectorizers
+nav-parent: Modules
+title: ref2vec-centroid
+description: Calculate vectors based on the centroid of references
+tags: ['ref2vec-centroid', 'centroid']
+menu-order: 7
+open-graph-type: article
+toc: true
+redirect_from:
+    - /developers/weaviate/current/modules/ref2vec-centroid.html
+---
+
+# Introduction
+
+The `ref2Vec-centroid` module is used to calculate vectors based on the centroid of existing vectors. The idea is that this centroid vector would be calculated from the vectors of an object's references, enabling associations between clusters of objects. This is useful in applications such as making suggestions based on the aggregation of a user's actions, or preferences.
+
+# How to enable
+
+## Weaviate Cloud Service
+
+The `ref2vec-centroid` module is not currently available on the WCS.
+
+## Weaviate open source
+
+Which modules to use in a Weaviate instance can be specified in the docker-compose configuration file. The service can be added like this:
+
+```yaml
+---
+version: '3.4'
+services:html
+  weaviate:
+    command:
+    - --host
+    - 0.0.0.0
+    - --port
+    - '8080'
+    - --scheme
+    - http
+    image: semitechnologies/weaviate:{{ current_page_version | remove_first: "v" }}
+    ports:
+    - 8080:8080
+    restart: on-failure:0
+    environment:
+      QUERY_DEFAULTS_LIMIT: 25
+      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
+      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
+      ENABLE_MODULES: 'ref2vec-centroid'
+      CLUSTER_HOSTNAME: 'node1'
+...
+```
+
+# How to configure
+
+​In your Weaviate schema, you must define how you want this module to vectorize your data. If you are new to Weaviate schemas, you might want to check out the [getting started guide on the Weaviate schema](../getting-started/schema.html) first.
+
+For example, here is an `Article` class which is configured to use ref2vec-centroid. This module requires only a class-level `moduleConfig`, containing two fields:
+
+1. `referenceProperties`: a list of the class' reference properties which should be used during the calculation of the centroid.
```

Comment:
"class'" is there an s missing?

Reply:
Both `Class's` and `Class'` are correct ways to express a singular possessive noun which ends in an `s`  🙂

Reply:
ha, you never stop learning :)